Scripts/New-Release.ps1

```diff
@@ -0,0 +1,152 @@
+<#
+
+.SYNOPSIS
+Creates a new release of the Hyper project.
+    
+.PARAMETER ReleaseName
+Name of the release.
+
+.PARAMETER MainBranchCheck
+Indicates whether to check if the release is being done from the main branch.
+
+.PARAMETER CreateSourceArchive
+Indicates whether to create an archive with the source code.
+
+.EXAMPLE
+
+PS> ./Scripts/New-Release.ps1 -ReleaseName Hyper-v0.0.0
+
+.EXAMPLE
+
+PS> ./Scripts/New-Release.ps1 -ReleaseName Hyper-v0.0.0 -CreateSourceArchive $false -MainBranchCheck $false
+
+#>
+
+param(
+    [Parameter(Mandatory)]
+    [string]$ReleaseName,
+    [bool]$MainBranchCheck = $true,
+    [bool]$CreateSourceArchive = $true
+)
+
+function CheckIfMainBranchIsTracked {
+    $gitBranchOut = @(git branch -vv)
+    foreach ($branch in $gitBranchOut) {
+        if ($branch -match "\* main\s+\w+ \[origin\/main].*") {
+            return $true
+        }
+    }
+    $false
+}
+
+function GitCheck {
+    # make sure main is the current branch
+    $currentBranch = git rev-parse --abbrev-ref HEAD
+    if ($currentBranch -cne "main") {
+        Write-Host "You're not on the main branch."
+        exit
+    }
+
+    # check if main is tracked
+    if ((CheckIfMainBranchIsTracked) -eq $false) {
+        Write-Host "Main not tracked."
+        exit
+    }
+
+    # update refs
+    git fetch
+
+    if ($LASTEXITCODE -ne 0) {
+        Write-Host "Couldn't fetch from origin main"
+        exit
+    }
+
+    # check if main is up-to-date
+    $gitStatus = git status -uno
+    $expectedGitStatusOutput = "Your branch is up to date with 'origin/main'."
+
+    if ($gitStatus[1] -cne $expectedGitStatusOutput) {
+        Write-Host "Local main not up-to-date with the remote."
+        exit
+    }
+}
+
+$originalDirectory = Get-Location
+$scriptDirectory = Split-Path -Parent $MyInvocation.MyCommand.Definition
+$rootDirectory = Split-Path -Parent $scriptDirectory
+$publishDirectory = "publish"
+$publishPath = Join-Path -Path $rootDirectory -ChildPath $publishDirectory # | Join-Path -ChildPath $ReleaseName
+
+Set-Location $rootDirectory
+
+$configuration = "Release"
+$runtimes = @("win-x64", "win-x86")
```

Comment:
don't we want to add linux?

Reply:
Will it work on linux? I have no way to test that